homeassistant/components/forecast_solar/diagnostics.py

```diff
@@ -0,0 +1,44 @@
+"""Diagnostics support for Forecast.Solar integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.diagnostics import async_redact_data
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY, CONF_LATITUDE, CONF_LONGITUDE
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN
+
+TO_REDACT = {
+    CONF_API_KEY,
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+}
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant, entry: ConfigEntry
+) -> dict[str, Any]:
+    """Return diagnostics for a config entry."""
+    coordinator: DataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    return {
+        "entry": {
+            "title": entry.title,
+            "data": async_redact_data(entry.data, TO_REDACT),
+            "options": async_redact_data(entry.options, TO_REDACT),
+        },
+        "data": {
```

Comment:
We should dump wh_days/wh_hours/watts as well, those will actually help to debug when things are not correct. The ones below are the results of those as well.

Reply:
What is the best way to dump that? Because of the `datetime` values

Reply:
There is not really a good way to handle that. 
I would just iterate over those dictionaries and call isostring() on those datetime objects.

Reply:
Lets finish this PR too, but I just can't seem to iterate over the suggested variables :(

Reply:
Added a commit, please check and let me know what you think 👍

Reply:
Exactly the result we were looking for 😄